dns_sd.h

```diff
@@ -15,11 +15,15 @@
 
 #ifdef _WIN32
 #include <winsock2.h>
+#include <ntddndis.h>
+#include <netioapi.h>
 #else
+#include <net/if.h>
 #include <sys/param.h>
 #endif
 
-#define DNS_SD_ADDRESS_STR_MAX (40) /* IPv6 Max = 4*8 + 7 + 1 for NUL */
+/* IPv6 Max = 4*8 + 7 + 1 for NUL + 1 for '%' + interface length */
+#define DNS_SD_ADDRESS_STR_MAX (41 + IF_NAMESIZE)
```

Comment:
`IF_NAMESIZE` already includes the null byte the way I read [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/net_if.h.html) which states "_symbolic constant for the length of a buffer containing an interface name (including the terminating NULL character)_".